pkg/configprovider/providerutil/utils.go

```diff
@@ -0,0 +1,45 @@
+package providerutil
+
+import (
+	"go.opentelemetry.io/collector/confmap"
+	"strings"
+)
+
+func removeKeyFromSrcMap(srcMap map[string]interface{}, mergeMap map[string]interface{}, keys []string, index int) map[string]interface{} {
+
+	if index == len(keys) { //out of index
+		return srcMap
+	}
```

Comment:
Another approach is to pass `keys[1:]` in the next iteration and if `len(keys) == 1` then it means this is the leaf key. I think it will be maybe less performant

Reply:
It seems only references are passed around when using keys[1:], so no actual performance impact, using this method as we don't have to pass around indexes